scripts/TPI.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+import scipy.optimize as opt
+
+def HHFOCs(b_sp1, *params):
+    '''
+    Refer to the equations on chapter 8
+    '''
+    BQpath, rho_s, beta, sigma, b_init, n, rpath, wpath = params
+    # rpath = length p, n = length p, p = number of periods left in life time, wpath = length p, b_sp1 = length p-1
+    
+    b_s = np.append(b_init, b_sp1)
+    b_sp1 = np.append(b_sp1, 0.0)
+    c = (1 + rpath) * b_s + wpath * n + BQpath - b_sp1
+    MU_c = c ** (-sigma)
+    errors = (MU_c[:-1] - beta * (1 + rpath[1:]) *
+              (1 - rho_s[:-1]) * MU_c[1:])
+    
+    return errors
+
+
+
```

Comment:
I'd suggest to define this as a function so that we can use it in the execute.py